app/components/donate-form.jsx

```diff
@@ -0,0 +1,218 @@
+import { CardElement, injectStripe } from 'react-stripe-elements';
+import { Component }                 from 'react';
+import { Link }                      from 'react-router';
+import { connect }                   from 'react-redux';
+
+import { AlertComponent }  from './alert';
+import { ReactGA }         from '../utils/analytics';
+import { checkVersion }    from '../actions/utils';
+import { dollar, padding } from '../utils/formatting';
+import { donate }          from '../actions/donation';
+
+export class DonateForm extends Component {
+
+  constructor (props) {
+    super(props);
+    this.state = {
+      amount: '1',
+      error: null,
+      loading: false,
+      months: Reflect.apply(Array, null, Array(12)).map((_, i) => padding(i + 1, 2)),
+      success: false,
+      years: Reflect.apply(Array, null, Array(10)).map((_, i) => new Date().getUTCFullYear() + i)
+    };
+  }
+
+  componentWillMount () {
+    const { checkVersion } = this.props;
+
+    checkVersion();
+  }
+
+  scrollToTop () {
+    if (this._form) {
+      this._form.scrollTop = 0;
+    }
+  }
+
+  donate = (e) => {
+    e.preventDefault();
+
+    const { donate, stripe } = this.props;
+    const { amount } = this.state;
+    const name = this._name.value || undefined;
+    const email = this._email.value;
+    const message = this._message.value || undefined;
+    const otherAmount = amount === 'other' && this._otherAmount.value;
+
+    this.setState({ error: null, loading: true });
+
+    stripe.createToken({ name })
+    .then(({ error, token }) => {
+      if (error) {
+        throw error;
+      }
+
+      return donate({
+        name,
+        email,
+        token: token.id,
+        message,
+        amount: amount === 'other' ? parseFloat(otherAmount.replace(/[^\d\.]/g, '')) : amount
+      });
+    })
+    .then(() => {
+      ReactGA.event({ action: 'donate', category: 'Donation' });
+
+      this.setState({ loading: false, success: true });
+    })
+    .catch((err) => {
+      this.setState({ error: err.message, loading: false });
+      this.scrollToTop();
+    });
+  }
+
+  render () {
+    const { session } = this.props;
+    const { amount, error, loading, success } = this.state;
+
+    let header = null;
+    let successNote = null;
+
+    if (session) {
+      successNote = (
+        <p>We'll be adding your profile to our project's <a href="https://github.com/pokedextracker/pokedextracker.com" target="_blank" rel="noopener noreferrer" className="link">list of supporters</a> shortly (if you're not already there). P.S.&mdash;check out your <Link className="link" to={`/u/${session.username}`}>new profile flair</Link>!</p>
+      );
+    }
+
+    if (success) {
+      header = (
+        <h1>Thanks for the donation! 💕</h1>
```

Comment:
didn't know what the best way to do an emoji was... i just c/p-ed the unicode character. lmk if there's a better way...